include/ignition/gazebo/gui/GuiSystem.hh

```diff
@@ -53,6 +54,14 @@ namespace gazebo
     /// and write entities and their components.
     public: virtual void Update(const UpdateInfo &/*_info*/,
                                 EntityComponentManager &/*_ecm*/){}
+
+    /// \brief When running in the same process GUI and server we need to
+    /// configure the gui system plugins with the event manager to listen
+    /// ecm updates and set if we are running in the same process
+    /// \param[in] _ecm Mutable event manager
+    /// \param[in] _sameProcess is server and gui running in the same process ?
+    public: virtual void Configure(
+      EventManager &/*_event*/, bool /*sameProcess*/){}
```

Comment:
As discussed in https://github.com/ignitionrobotics/ign-gazebo/pull/793#discussion_r654643454, would it be possible to use a component instead of having this method being used in all GUI systems?

Reply:
I made this for `sameProcess` in this commit https://github.com/ignitionrobotics/ign-gazebo/pull/793/commits/9962dbf826132a6d777792ecf2715d51b09e0786

But I don't know if we can do it with the `EventManager` because we need the reference and I believe components are just a copy, isn't it ?

Reply:
> components are just a copy, isn't it ?

I believe so - usually, the data in a component is instantiated via the `=` operator, which results in a copy unless the type of data being assigned has the `=` operator overwritten to avoid copies. But, we haven't done this for all component data types, so I believe it's a copy.